feat: Allow to add more org file locations (#42)

seflue opened the pull request:
Supporting more then one org file locations allows the user to integrate existing orgmode agenda files into the org-roam knowledge graph.

Implements #42.

chipsenkbeil commented on DOCS.org:
```diff
@@ -59,17 +59,56 @@
    #+end_src
 
+   Configuring additional folders and files to be loaded by org-roam is also
+   supported, see [[#configuration-org_files][org_files]]:
```
Nitpick: replace the `:` with `.` here. 😄

chipsenkbeil commented on DOCS.org:
```diff
@@ -59,17 +59,56 @@
+   #+end_src
+
+   To actually connect the content of these files with org-roams knowledge
+   graph, you can use the "store link" feature of orgmode (=,ols=), which adds
```
You have the keybinding as `,ols` as I assume the default leader key is comma. Can we show it as `<Leader>ols` instead?

seflue replied:
You are absolutely correct. I confused myself here, actually I have the prefix of orgmode mapped to just `,` and the default prefix is `<Leader>o`. `,o` is completely wrong, it is a mash-up of my mapping and the default one. :sweat_smile:

chipsenkbeil commented on DOCS.org:
```diff
@@ -59,17 +59,56 @@
+   #+end_example
+
+   Tip: Because "store link" in orgmode currently does not support whole files, the
+   quick workaround is to apply =,ols= on a headline and move the =:PROPERTIES:=
```
Same here. Can we show the keybinding as `<Leader>ols` instead?

chipsenkbeil commented on lua/org-roam/database.lua:
```diff
@@ -94,6 +98,10 @@ function M:files_path()
     return self.__directory
 end
 
+function M:org_files()
```
Since this is a publicly-exposed function, would you mind giving it a Luadoc comment? For example, it would be good to note that the files returned are globs and not necessarily individual files or directories.

chipsenkbeil commented on spec/setup_spec.lua:
```diff
@@ -29,4 +29,77 @@ describe("org-roam.setup", function()
+        return true
+    end
+
+    local say = require("say")
```
My own naiveness here. Where does this library come from? Is this something that plenary.nvim provides? Or does it come in from somewhere else? And I don't actually know what say does. Can you explain, maybe adding a comment section above, what this is doing so it's documented for future test contributions?

seflue replied:
I actually also wondered, when I saw the example code from luassert's `register` function (luassert is the asserting library deployed with busted). I then tried it out and it just works. Turns out it is a hashing library and key-value store used by busted and luassert and maintained - also like busted and luassert - in the lunarmodules project, see [here](https://github.com/lunarmodules/say).

seflue replied:
I added a comment as suggested by you.

chipsenkbeil reviewed (changes requested):
Looks fine overall. Left some requests that are nitpicks on grammar and asks for some documentation. Once we get that updated, let's merge this in as another fantastic contribution from your side! 😄

chipsenkbeil commented:
Oh, and I wasn't sure if you were done with the branch from your last PR. Since it seems like you are, we should delete that branch and this one once it's merged.

seflue commented:
> Looks fine overall. Left some requests that are nitpicks on grammar and asks for some documentation. Once we get that updated, let's merge this in as another fantastic contribution from your side! 😄

I addressed all your comments, fixed the typos and mistakes and added the documentation and comments.

> Oh, and I wasn't sure if you were done with the branch from your last PR. Since it seems like you are, we should delete that branch and this one once it's merged.

Yes, we can delete the last branch and also this one as soon as the PR is merged.

chipsenkbeil commented:
Good to go!
